### PR TITLE
Transaction impl to spec

### PR DESCRIPTION
**Pull request by MavenRain**
This PR refactors Transaction in the many-types module to match specification, as cited in https://github.com/liftedinit/many-rs/issues/207

### CODE REVIEW

**Review by fmorency — APPROVED** [review]
LGTM, thanks!

Wait for the `many-framework` work to be completed before merging this.

**Review by fmorency — CHANGES_REQUESTED** [review]
We might want to burn the content index (`#[n(1)]`) as per 

https://github.com/many-protocol/specification/blob/main/spec/deprecation.adoc

File: src/many-types/src/blockchain.rs [hunk 1]
```diff
@@ -83,8 +83,11 @@ pub struct Transaction {
     #[n(0)]
     pub id: TransactionIdentifier,
 
-    #[cbor(n(1), with = "minicbor::bytes")]
-    pub content: Option<Vec<u8>>,
+    #[n(2)]
```
Comment (MavenRain) [inline src/many-types/src/blockchain.rs:86]: See https://github.com/many-protocol/specification/pull/83 for an explanation of this index shift.

**Review by fmorency — CHANGES_REQUESTED** [review]
You need to keep the legacy `Transaction` struct somewhere, as you'll need it for the migration.

Check how the `Memo` refactoring was made, especially
- many-types/src/memo.rs
- many-types/src/memo/legacy.rs

**Review by fmorency — CHANGES_REQUESTED** [review]
You don't need the `merk`-related stuff in `many-error`?

Some minor changes to mimic how the `memo` refactor.

File: README.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 [![coverage](https://img.shields.io/codecov/c/gh/liftedinit/many-rs)](https://app.codecov.io/gh/liftedinit/many-rs)
 [![license](https://img.shields.io/github/license/liftedinit/many-rs)](https://github.com/liftedinit/many-rs/blob/main/LICENSE)
 
-Rust libraries for the [MANY protocol](https://github.com/many-protocol).
+Rust libraries for the [MANY protocol] can be found in (https://github.com/many-types).
```
Comment (hansl) [inline README.md:6]: Revert this. This should be a link to the specification.

**Review by fmorency — CHANGES_REQUESTED** [review]
Wait a minute @MavenRain...

This is wrong - you didn't follow the specification.

```
? 1 => bstr .cbor request / bstr .size 0,
? 2 => bstr .cbor response / bstr .size 0,
```

**Comment by MavenRain** [comment]
Closing in favour of https://github.com/liftedinit/many-rs/pull/234

**Pull request closed without merging**